plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMinioConnectorSmokeTest.java

```diff
@@ -138,8 +143,65 @@ public void testMetadataLocationWithDoubleSlash()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testExpireSnapshotsBatchDeletes()
+    {
+        String tableName = "test_expiring_snapshots_" + randomTableSuffix();
+        Session sessionWithShortRetentionUnlocked = prepareCleanUpSession();
+        String location = "s3://%s/%s/%s/".formatted(bucketName, schemaName, tableName);
+        List<Event> events = new ArrayList<>();
+        hiveMinioDataLake.getMinioClient().captureBucketNotifications(bucketName, event -> {
+            if (event.eventType().toString().toLowerCase(ENGLISH).contains("remove")) {
+                events.add(event);
```

Comment:
is `events` access happening in one thread always?

Reply:
good point I don't know, I will make it thread safe